flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -376,7 +381,7 @@ public int getBuffersInBacklog() {
 
 	private boolean shouldNotifyDataAvailable() {
 		// Notify only when we added first finished buffer.
-		return readView != null && !flushRequested && getNumberOfFinishedBuffers() == 1;
+		return readView != null && !flushRequested && !readView.isBlockedByCheckpoint() && getNumberOfFinishedBuffers() == 1;
```

Comment:
also we can remove this condition by reusing it inside `notifyDataAvailable` as mentioned above.

Reply:
Some tags should be checked in lock, I removed the check in notifyDataAvailable.